script.js

```diff
@@ -1,12 +1,169 @@
 //You can edit ALL of the code here
+
+let allEpisodes = getAllEpisodes();
 function setup() {
-  const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
 }
 
+// headerContainer.classList.add("header-container");
+//Level 350 - Switch to fetching live data!
+function fetchAllEpisodes(series) {
+  return fetch(`https://api.tvmaze.com/shows/${series}/episodes`)
+    .then((response) => response.json())
+    .then((data) => {
+      allEpisodes = data;
+      makePageForEpisodes(allEpisodes);
+    })
+    .catch(function (error) {
+      console.log(error);
+    });
+}
+
+function tvSeriesSetup(showList) {
+  let selectSeries = document.getElementById("show-select");
+  let optionSeries = document.createElement("option");
+  optionSeries.textContent = "Please Choose Your Show";
+  selectSeries.appendChild(optionSeries);
+  selectSeries.selectedIndex = 0;
+
+  //sorting based on alphabetical order
+  showList.sort(function (a, b) {
+    return a.name.localeCompare(b.name);
+  });
+  for (let i = 0; i < showList.length; i++) {
+    let option = document.createElement("option");
+    option.value = showList[i].id;
+    option.textContent = `${showList[i].name}`;
+    selectSeries.appendChild(option);
+  }
+  selectSeries.addEventListener("change", function () {
+    console.log(selectSeries.value);
+    fetchAllEpisodes(selectSeries.value);
+  });
+}
+//its optional will refactor later
+function setup() {
+  const seriesList = getAllShows();
+  tvSeriesSetup(seriesList);
+}
+//creating search bar//
+const searchElem = document.querySelector("#search");
+searchElem.addEventListener("input", searchEpisode);
+function searchEpisode() {
+  const searchInput = searchElem.value.toLowerCase();
+  const filteredEpisodes = allEpisodes.filter((episode) => {
+    if (
+      episode.name.toLowerCase().includes(searchInput) ||
+      episode.summary.toLowerCase().includes(searchInput)
```

Comment:
This line is throwing an uncaught error sometimes. I think not all shows have summary. 

```
script.js:55 Uncaught TypeError: Cannot read properties of null (reading 'toLowerCase')
    at script.js:55:77
    at Array.filter (<anonymous>)
    at HTMLInputElement.searchEpisode (script.js:54:40)
(anonymous) @ script.js:55
searchEpisode @ script.js:54
```

You can add conditional chaining to avoid such errors: `episode.summary?.toLowerCase()`

Reply:
i have updated code to following:
`const filterEpisodes = allEpisodes.filter((episode) => {
    if (
      (episode.name && episode.name.toLowerCase().includes(searchInput)) ||
      (episode.summary && episode.summary.toLowerCase().includes(searchInput))
    ) {
      return true;
    `}`
    return false;
  });`
and it seems working. thank you